proc-macros/tests/ui/correct/rpc_deny_missing_docs.rs

```diff
@@ -13,7 +13,7 @@ pub trait ApiWithDocumentation {
 
 	/// Subscription docs.
 	#[subscription(name = "sub", unsubscribe = "unsub", item = String)]
-	fn sub(&self);
+	async fn sub(&self) -> jsonrpsee::core::SubscriptionResult;
```

Comment:
I'm not sure whether we should force folks to return `SubscriptionResult` in the proc macro API.

The generated client trait returns another type anyway.